src/librustc_resolve/lib.rs

```diff
@@ -2283,24 +2283,25 @@ impl<'a> Resolver<'a> {
                 PatKind::Ident(bmode, ref ident, ref opt_pat) => {
                     // First try to resolve the identifier as some existing
                     // entity, then fall back to a fresh binding.
-                    let resolution = if let Ok(resolution) = self.resolve_path(pat.id,
-                                &Path::from_ident(ident.span, ident.node), 0, ValueNS) {
+                    let local_def = self.resolve_identifier(ident.node, ValueNS, true);
```

Comment:
Something is wrong here. Ideally, the adjustment routine should not report "can't capture" errors, it should just leave `Local`s outside of closures unadjusted.
Regarding
```
fn f(x: i32) {
    fn g(x: i32) {}
}
```
the consensus in https://github.com/rust-lang/rust/issues/33118 seemed to be that items in block scopes are isolated from local variables, i.e. `g` shouldn't "see" the outer `x` and the inner `x` won't resolve to it.